File: solutions/python/matrix/1/matrix.py

class Matrix:
   
    def __init__(self, matrix_string):
        matrix_list=[]
        r=0
        for row_str in matrix_string.split('\n'):
            r=r+1
            matrix_list.append([])
            c=0
            for col_str in row_str.split(' '):
                c=c+1
                matrix_list[r-1].append(int(col_str))
        self.r=r
        self.c=c
        self.values = matrix_list

    def row(self, index):
        return(self.values[index-1])
        

    def column(self, index):
        col = []
        for row in range(self.r):
            col.append(self.values[row][index-1])
        return(col)

File: solutions/fortran/matrix/1/matrix.f90

module matrix
  implicit none

contains

  function row(matrix, dims, i) result(r)
    integer, dimension(2), intent(in) :: dims
    !! Matrix dimensions (nrows, ncols)
    character(len=*), dimension(dims(1)), intent(in) :: matrix
    !! Matrix as a 1-d array of strings
    integer, intent(in) :: i
    !! Row index
    integer, dimension(dims(2)) :: r
    integer :: stat
    r(:) = 0
    read(matrix(i), *, IOSTAT=stat) r
  end function

  function column(matrix, dims, j) result(c)
    integer, dimension(2), intent(in) :: dims
    !! Matrix dimensions (nrows, ncols)
    character(len=*), dimension(dims(1)), intent(in) :: matrix
    !! Matrix as a 1-d array of strings
    integer, intent(in) :: j
    !! Column index
    integer, dimension(dims(1)) :: c
    integer, dimension(dims(2)) :: r
    integer :: k
    integer :: stat
    c(:) = 0
    do k=1,dims(1)
      read(matrix(k),*,IOSTAT=stat) r
      c(k) = r(j)
    end do
  end function

end module
